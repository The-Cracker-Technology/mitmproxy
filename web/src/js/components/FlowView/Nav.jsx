import React  from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import classnames from 'classnames'
import _ from 'lodash'

NavAction.propTypes = {
    icon: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
}

export function NavAction({ icon, title, onClick }) {
    return (
        <a title={title}
            href="#"
            className="nav-action"
            onClick={event => {
                event.preventDefault()
                onClick(event)
            }}>
            <i className={`fa fa-fw ${icon}`}></i>
        </a>
    )
}

Nav.propTypes = {
    active: PropTypes.string.isRequired,
    tabs: PropTypes.array.isRequired,
    onSelectTab: PropTypes.func.isRequired,
}

export default function Nav({ active, tabs, onSelectTab }) {
    return (
        <nav className="nav-tabs nav-tabs-sm">
            {tabs.map(tab => (
                <a key={tab}
                    href="#"
                    className={classnames({ active: active === tab })}
                    onClick={event => {
                        event.preventDefault()
                        onSelectTab(tab)
                    }}>
                    {_.capitalize(tab)}
                </a>
            ))}
        </nav>
    )
}
