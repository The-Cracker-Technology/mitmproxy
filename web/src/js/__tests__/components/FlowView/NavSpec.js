import React from 'react'
import renderer from 'react-test-renderer'
import Nav, { NavAction } from '../../../components/FlowView/Nav'

describe('Nav Component', () => {
    let tabs = ['foo', 'bar'],
        onSelectTab = jest.fn(),
        nav = renderer.create(<Nav active='foo' tabs={tabs} onSelectTab={onSelectTab}/>),
        tree = nav.toJSON()

    it('should render correctly', () => {
        expect(tree).toMatchSnapshot()
    })

    it('should handle click', () => {
        let mockEvent = { preventDefault: jest.fn() }
        tree.children[0].props.onClick(mockEvent)
        expect(mockEvent.preventDefault).toBeCalled()
        expect(onSelectTab).toBeCalledWith('foo')
    })
})

describe('NavAction Component', () => {
    let clickFn = jest.fn(),
        navAction = renderer.create(<NavAction icon="foo" title="bar" onClick={clickFn}/>),
        tree = navAction.toJSON()

    it('should render correctly', () => {
        expect(tree).toMatchSnapshot()
    })

    it('should handle click', () => {
        let mockEvent = { preventDefault: jest.fn() }
        tree.props.onClick(mockEvent)
        expect(mockEvent.preventDefault).toBeCalled()
        expect(clickFn).toBeCalledWith(mockEvent)
    })
})
