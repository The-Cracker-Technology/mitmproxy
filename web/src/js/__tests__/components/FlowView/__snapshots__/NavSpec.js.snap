// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component should render correctly 1`] = `
<nav
  className="nav-tabs nav-tabs-sm"
>
  <a
    className="active"
    href="#"
    onClick={[Function]}
  >
    Foo
  </a>
  <a
    className=""
    href="#"
    onClick={[Function]}
  >
    Bar
  </a>
</nav>
`;

exports[`NavAction Component should render correctly 1`] = `
<a
  className="nav-action"
  href="#"
  onClick={[Function]}
  title="bar"
>
  <i
    className="fa fa-fw foo"
  />
</a>
`;
